cookbook/product_asset/add_new_transformation.rst

```diff
@@ -0,0 +1,243 @@
+        }
+
+        /**
+         * {@inheritdoc}
+         */
+        public function getName()
+        {
+            return 'watermark';
+        }
+    }
+
+.. _ImageMagickLauncher: https://github.com/akeneo/pim-enterprise-dev/blob/master/src/Akeneo/Component/FileTransformer/Transformation/Image/ImageMagickLauncher.php
+.. note::
+    To see how the imagemagick command is launched, you can take a look to the ImageMagickLauncher_.
+
+In order to rely on options you can add an OptionsResolver, for this you need to follow this rule :
+
+ * your OptionsResolver must implement the ``Akeneo\Component\FileTransformer\Options\TransformationOptionsResolverInterface``.
+
+.. code-block:: php
+
+    #/src/Akeneo/Component/FileTransformer/Options/Image/WatermarkOptionsResolver.php
+    <?php
+
+    namespace Akeneo\Component\FileTransformer\Options\Image;
+
+    use Akeneo\Component\FileTransformer\Exception\InvalidOptionsTransformationException;
+    use Akeneo\Component\FileTransformer\Options\TransformationOptionsResolverInterface;
+    use Symfony\Component\OptionsResolver\OptionsResolver;
+
+    /**
+     * Watermark Options Resolver
+     */
+    class WatermarkOptionsResolver implements TransformationOptionsResolverInterface
+    {
+        /** @var OptionsResolver */
+        protected $resolver;
+
+        public function __construct()
+        {
+            $this->resolver = new OptionsResolver();
+            $this->resolver->setRequired([
+                'font',
+                'size',
+                'watermark',
+                'positionBlack',
+                'positionWhite',
+            ]);
+            $this->resolver->setAllowedTypes([
+                'font'          => 'string',
+                'size'          => 'string',
+                'watermark'     => 'string',
+                'positionBlack' => 'string',
+                'positionWhite' => 'string',
+            ]);
+            $this->resolver->setDefaults([
+                'font'          => 'Arial',
+                'size'          => '12',
+                'watermark'     => 'Copyright',
+                'positionBlack' => '0,45',
+                'positionWhite' => '1,43',
+            ]);
+        }
+
+        /**
+         * {@inheritdoc}
+         */
+        public function resolve(array $options)
+        {
+            try {
+                $options = $this->resolver->resolve($options);
+            } catch (\Exception $e) {
+                throw InvalidOptionsTransformationException::general($e, 'watermark');
+            }
+
+            return $options;
+        }
+    }
+
+.. _OptionsResolver: http://symfony.com/doc/current/components/options_resolver.html
+.. note::
+    You can know more about this Symfony2 component on the OptionsResolver_ documentation.
+
+Next step is to add your services in existing files or in your own one :
+
+.. code-block:: yaml
+
+    # src/Akeneo/Bundle/FileTransformerBundle/Resources/config/options_resolvers.yml
```

Comment:
That's the real file. I can help users to see where are the others services and parameters. If you want I can't mention this file before and change the file name.